panel-plugin/i3w-config.c

```diff
@@ -58,14 +58,14 @@ config_dialog_closed(GtkWidget *dialog, int response, ConfigDialogClosedParam *p
 /* Function Implementations */
 
 guint32
-serialize_gdkcolor(GdkColor *gdkcolor)
+serialize_gdkrgba(GdkRGBA *gdkrgba)
 {
     guint32 color = 0;
 
-    // truncate the GdkColor components to 8 bit
-    guint16 red_component = gdkcolor->red >> 8;
-    guint16 green_component = gdkcolor->green >> 8;
-    guint16 blue_component = gdkcolor->blue >> 8;
+    // convert the GdkRGBA components to 8 bit ints
+    guint8 red_component = gdkrgba->red * 255;
```

Comment:
`* 255` is not equivalent to `>> 8`, is this a difference between the two APIs?

Reply:
Yup! GdkRGBA stores its components in floats of range 0.0 to 1.0. Multiplying and dividing by 255 (max of a uint8) converts to/from 8 bit ints, whereas shifts will do interesting (incorrect) things.

I also have a branch (https://github.com/ad1217/xfce4-i3-workspaces-plugin/tree/gtk3-css) which I wrote last night that does away with this part entirely in favor of CSS styling, which seems to be the preferred method for GTK3 and simplifies a lot of the color-related code.

Reply:
Your branch works for me in Arch. I've not noticed any bugs, just that only the text is coloured, rather than the coloured boxes as in the original. If you're planning to maintain this package for the future, is it beyond your scope to add a few features? It would be nice to be able to change the font and customise the background, border, and text colours for each workspace type, to make it look like the native i3 bar. Otherwise I'm just happy that functionality is restored, thanks!

Reply:
@ad1217 I'm going to merge this branch as-is, and you can submit your CSS branch as a follow-up (if you want to).